python/dask_cudf/dask_cudf/tests/test_core.py

```diff
@@ -782,14 +782,16 @@ def test_dataframe_set_index():
     df["str"] = list("abcdefghijklmnopqrstuvwxyz")
     pdf = df.to_pandas()
 
-    ddf = dgd.from_cudf(df, npartitions=4)
-    ddf = ddf.set_index("str")
+    with dask.config.set({"dataframe.convert-string": False}):
+        ddf = dgd.from_cudf(df, npartitions=4)
+        ddf = ddf.set_index("str")
 
-    pddf = dd.from_pandas(pdf, npartitions=4)
-    pddf = pddf.set_index("str")
-    from cudf.testing._utils import assert_eq
+        pddf = dd.from_pandas(pdf, npartitions=4)
+        pddf = pddf.set_index("str")
+
+        from cudf.testing._utils import assert_eq
```

Comment:
Note for future work:

`dd.assert_eq` attempts to convert both sides to pyarrow strings after converting to pandas, `cudf.testing._utils import assert_eq` does not do this. If the cudf version is revised, this test will pass without the "dataframe.convert-string" change.